### ISSUE TITLE
Bug in `multiline-ternary` rule with "always-multiline" option

### ISSUE THREAD

**Issue by feross** [eslint/eslint#13782]
**Tell us about your environment**

ESLint Version: v7.11.0
Node version: v14.14.0
npm version: v6.14.8

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
    "multiline-ternary": ["error", "always-multiline"],
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

This returns an error:

```js
      this._address = {
        port: info.localPort,
        family: info.localAddress &&
          info.localAddress.indexOf(':') !== -1 ? 'IPv6' : 'IPv4',
        address: info.localAddress
      }
```

However, this does not:

```js
      this._address = {
        port: info.localPort,
        family: info.localAddress && info.localAddress.indexOf(':') !== -1 ? 'IPv6' : 'IPv4',
        address: info.localAddress
      }
```

**What did you expect to happen?**

No warning since the ternary is fully contained on a single line and the `always-multiline` option is selected.

**What actually happened? Please include the actual, raw output from ESLint.**

  /Users/feross/code/standard/tmp/chrome-net/index.js:336:17: Expected newline between test and consequent of ternary expression. (multiline-ternary)

**Are you willing to submit a pull request to fix this bug?**

Yes, but I'm not familiar with the rule in question.

**Comment by feross** [comment]
Oh, perhaps the rule is getting triggered because the "test" is actually on two lines:

```js
info.localAddress &&
          info.localAddress.indexOf(':') ...
```
